src/python/grpcio/grpc/_server.py

```diff
@@ -571,30 +596,35 @@ def _handle_with_method_handler(rpc_event, method_handler, thread_pool):
         if method_handler.request_streaming:
             if method_handler.response_streaming:
                 _handle_stream_stream(rpc_event, state, method_handler,
-                                      thread_pool)
+                                      thread_pool, server_state)
             else:
                 _handle_stream_unary(rpc_event, state, method_handler,
-                                     thread_pool)
+                                     thread_pool, server_state)
         else:
             if method_handler.response_streaming:
                 _handle_unary_stream(rpc_event, state, method_handler,
-                                     thread_pool)
+                                     thread_pool, server_state)
             else:
                 _handle_unary_unary(rpc_event, state, method_handler,
-                                    thread_pool)
+                                    thread_pool, server_state)
         return state
 
 
-def _handle_call(rpc_event, generic_handlers, thread_pool):
+def _handle_call(rpc_event, generic_handlers, thread_pool, server_state):
     if not rpc_event.success:
         return None
     if rpc_event.request_call_details.method is not None:
         method_handler = _find_method_handler(rpc_event, generic_handlers)
-        if method_handler is None:
+        if (server_state.max_requests is not None and
+                server_state.outstanding_requests >= server_state.max_requests):
+            return _reject_call(rpc_event)
+        elif method_handler is None:
             return _handle_unrecognized_method(rpc_event)
         else:
+            # The server_state should be locked here
```

Comment:
Drop this; if we had something like [Java's `GuardedBy`](http://jcip.net/annotations/doc/net/jcip/annotations/GuardedBy.html) of course we'd use it but I don't think it's worth a prose implementation comment. `_handle_call` has one call site.